src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -407,6 +405,16 @@ internal static Collection<CustomAttributeBuilder> CreateTypeAttributes(ScriptHo
             return customAttributes;
         }
 
+        private void RestartHost()
+        {
+            // signal host restart
+            _restartEvent.Set();
+
+            // whenever we're restarting the host, we want to let the Node
+            // invoker know so it can clear the require cache, etc.
+            NodeFunctionInvoker.OnHostRestart();
```

Comment:
Would be nice to remove knowledge of the Node invoker from here. Can we maybe invert this and expose a static host event the invoker can subscribe to?